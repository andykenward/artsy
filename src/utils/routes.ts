import queryString from 'query-string';
import { isNil, pick, pickBy } from 'ramda';

const allowedFilters = ['color'];

interface Filter {
  color?: string;
}

export const validateFilterObject = (
  searchQuery: string | undefined
): Filter | null => {
  if (!searchQuery || searchQuery === '') return null;
  const filter = pick(allowedFilters, queryString.parse(searchQuery));
  return pickBy((v) => !isNil(v), filter);
};

export const getFilter = (
  searchQuery: string | undefined
): { color?: string } | null => {
  return validateFilterObject(searchQuery);
};
