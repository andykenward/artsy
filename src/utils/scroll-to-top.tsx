import { RouteComponentProps } from '@reach/router';
import React, { useEffect } from 'react';

export const ScrollToTop: React.FC<RouteComponentProps> = ({
  children,
  location,
}) => {
  const { pathname } = location!;
  useEffect(() => {
    window.scrollTo(0, 0);
  }, [pathname]);

  return <>{children}</>;
};
