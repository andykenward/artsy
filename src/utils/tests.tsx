import { injectGlobalStyles, Theme } from '@artsy/palette';
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from '@reach/router';
import { render } from '@testing-library/react';
import React from 'react';

export const { GlobalStyles } = injectGlobalStyles(`
  @import url("https://webfonts.artsy.net/all-webfonts.css")
`);

export const renderWithTheme = (element: React.ReactElement<any>) =>
  render(
    <Theme>
      <>
        <GlobalStyles />
        {element}
      </>
    </Theme>
  );

export const renderWithRouter = (
  element: React.ReactElement<any>,
  {
    route = '/',
    history = createHistory(createMemorySource((route = '/'))),
  } = {}
) => ({
  ...render(<LocationProvider history={history}>{element}</LocationProvider>),
  // adding `history` to the returned utilities to allow us
  // to reference it in our tests (just try to avoid using
  // this to test implementation details).
  history,
});
