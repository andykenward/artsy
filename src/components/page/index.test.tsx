import React from 'react';
import { Page } from '.';
import { renderWithTheme } from '../../utils/tests';

describe('<Page />', () => {
  it('renders correctly', () => {
    const { container } = renderWithTheme(<Page />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with children', () => {
    const { container, getByText } = renderWithTheme(<Page>Hello World</Page>);
    expect(getByText('Hello World')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
