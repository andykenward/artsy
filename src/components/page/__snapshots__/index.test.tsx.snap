// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> renders correctly 1`] = `
.c0 {
  min-height: 100vh;
}

<main
  class="c0"
/>
`;

exports[`<Page /> renders with children 1`] = `
.c0 {
  min-height: 100vh;
}

<main
  class="c0"
>
  Hello World
</main>
`;
